packages/nimble-components/src/table/specs/table-keyboard-navigation-hld.md

```diff
@@ -0,0 +1,215 @@
+
+When a row is focused:
+
+-   If the row can be expanded and is collapsed, pressing `ArrowRight` will expand the row
+-   If the row is expanded and can be collapsed, pressing `ArrowLeft` will collapse the row
+-   If the row cannot be expanded (or is already expanded), pressing `ArrowRight` will move focus to the first cell
+-   If the row cannot be collapsed (or is already collapsed), pressing `ArrowLeft` does nothing
+-   If `ArrowDown` or `ArrowUp` is pressed, the next or previous row will be focused (if a row is available in the given direction to move to)
+-   Action menus: All cell action menu buttons for the given row will be visible
+
+When a cell is focused:
+
+-   If the focus is on the first cell, pressing `ArrowLeft` will focus the entire row
+-   If the focus is on any cell other than the first cell, pressing `ArrowLeft` will move focus one cell to the left
+-   If the focus is on any cell other than the last, pressing `ArrowRight` will move focus to the next cell to the right
+-   If the focus is on the last cell, pressing `ArrowRight` does nothing
+-   Note: If a focused cell has multiple interactive elements for its contents, those elements will _not_ receive focus from pressing `ArrowLeft` or `ArrowRight`, and instead focus will be shifted to the appropriate neighboring cell (or do nothing if focus is at the extents of the row).
+-   If `ArrowDown` or `ArrowUp` is pressed, when there is an available row in the navigation direction, focus will move to the cell in the current focused column of the new row (next row or previous row, respectively)
+-   Action menus: The cell menu for the given cell will be visible. Any other action menu buttons for the row will be hidden as usual (unless the row is also mouse-hovered)
+
+To navigate by multiple rows/cells, when a row or cell is focused:
+
+-   `PageUp`: Navigate up by a table height's worth of rows. If a cell was selected, the matching cell (in the same column) of the new row will be selected.
+-   `PageDown`: Navigate down by a table height's worth of rows. If a cell was selected, the matching cell (in the same column) of the new row will be selected.
+-   `Home`: Navigate to the 1st row (if a row is selected), or the 1st cell of the row (if a cell is selected)
+-   `End`: Navigate to the last row (if a row is selected), or the last cell of the row (if a cell is selected)
+-   `Ctrl-Home`: Navigate to the first row. If a cell was selected, the matching cell (in the same column) of the new row will be selected.
+-   `Ctrl-End`: Navigate to the last row. If a cell was selected, the matching cell (in the same column) of the new row will be selected.
+
+Special case of focusing a cell:
+
+-   If a cell does not have an action menu, and only contains a single interactive/focusable element of the following types: button (and variants), switch, checkbox, or anchor/link: that child element will be focused _instead_ of the cell.
```

Comment:
Whose job will it be to decide what needs to be focused (i.e. the cell or a specific element) and then focus it?

Note: I'm not looking to get too far into the implementation details here. I'm mostly wondering if this is something the table/row/cell can handle, or if we are needing to introduce a new API for the table column and/or cell view.

Reply:
We might need a new API for the `TableCellView`, similarly to where we ended up with the `focusedRecycleCallback()`. I still need to do some prototyping in this area to get a better idea of what the implementation should be like.

Reply:
If there ends up being API impact to the table/ cell, we can update the HLD with those details in a followup PR.